test/scheduler/pickall_integration_test.go

```diff
@@ -0,0 +1,216 @@
+			},
+			Spec: clusterv1beta1.MemberClusterSpec{
+				Identity: rbacv1.Subject{
+					Kind:     "ServiceAccount",
+					APIGroup: "",
+					Name:     "admin",
+				},
+			},
+		}
+		Expect(hubClient.Create(ctx, &memberCluster)).To(Succeed(), "Failed to create member cluster")
+
+		// Mark this cluster as healthy.
+		Expect(hubClient.Get(ctx, types.NamespacedName{Name: newUnhealthyMemberClusterName}, &memberCluster)).To(Succeed(), "Failed to get member cluster")
+		memberCluster.Status.AgentStatus = []clusterv1beta1.AgentStatus{
+			{
+				Type: clusterv1beta1.MemberAgent,
+				Conditions: []metav1.Condition{
+					{
+						Type:               string(clusterv1beta1.AgentJoined),
+						Status:             metav1.ConditionTrue,
+						LastTransitionTime: metav1.NewTime(time.Now()),
+						Reason:             dummyReason,
+					},
+					{
+						Type:               string(clusterv1beta1.AgentHealthy),
+						Status:             metav1.ConditionTrue,
+						LastTransitionTime: metav1.NewTime(time.Now()),
+						Reason:             dummyReason,
+					},
+				},
+				LastReceivedHeartbeat: metav1.NewTime(time.Now()),
+			},
+		}
+		Expect(hubClient.Status().Update(ctx, &memberCluster)).To(Succeed(), "Failed to update member cluster status")
+	})
+
+	It("should create scheduled bindings for the newly recovered cluster", func() {
+		scheduledBindingsCreatedActual := scheduledBindingsCreatedForClustersActual(updatedHealthyClusters, updatedZeroScoreByCluster, crpName, policySnapshotName)
+		Eventually(scheduledBindingsCreatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to create the expected set of bindings")
+		Consistently(scheduledBindingsCreatedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Failed to create the expected set of bindings")
+	})
+
+	It("can mark a healthy cluster as unhealthy", func() {
+		memberCluster := &clusterv1beta1.MemberCluster{}
+		Expect(hubClient.Get(ctx, types.NamespacedName{Name: newUnhealthyMemberClusterName}, memberCluster)).To(Succeed(), "Failed to get member cluster")
+		memberCluster.Status.AgentStatus = []clusterv1beta1.AgentStatus{
+			{
+				Type: clusterv1beta1.MemberAgent,
+				Conditions: []metav1.Condition{
+					{
+						Type:               string(clusterv1beta1.AgentJoined),
+						Status:             metav1.ConditionTrue,
+						LastTransitionTime: metav1.NewTime(time.Now()),
+						Reason:             dummyReason,
+					},
+					{
+						Type:               string(clusterv1beta1.AgentHealthy),
+						Status:             metav1.ConditionTrue,
+						LastTransitionTime: metav1.NewTime(time.Now()),
+						Reason:             dummyReason,
+					},
+				},
+				LastReceivedHeartbeat: metav1.NewTime(time.Now()),
+			},
+		}
+		Expect(hubClient.Status().Update(ctx, memberCluster)).To(Succeed(), "Failed to update member cluster status")
+	})
+
+	It("should not remove binding for the cluster that just becomes unhealthy", func() {
```

Comment:
will you cover the 1) membercluster label is changed 2) membercluster is left in the next PR?

Reply:
As mentioned in the PR description, it's only part of the test.